sql/core/src/main/scala/org/apache/spark/sql/internal/sessionStateBuilders.scala

```diff
@@ -134,6 +135,14 @@ abstract class BaseSessionStateBuilder(
   }
 
   /**
+   * Interface exposed to the user for registering user-defined functions.
+   *
+   * Note 1: The user-defined functions must be deterministic.
+   * Note 2: This depends on the `functionRegistry` field.
+   */
+  protected def udf: UDFRegistration = new UDFRegistration(functionRegistry)
```

Comment:
This file only contains effectively one builder. So it should be named after the class.